java/client/src/main/java/glide/api/logging/Logger.java

```diff
@@ -0,0 +1,180 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.logging;
+
+import lombok.Getter;
+import lombok.NonNull;
+
+/**
+ * A singleton class that allows logging which is consistent with logs from the internal rust core.
+ * The logger can be set up in 2 ways - <br>
+ * 1. By calling <code>Logger.init</code>, which configures the logger only if it wasn't previously configured.<br>
+ * 2. By calling <code>Logger.setLoggerConfig</code>, which replaces the existing configuration, and means that new logs will not be
+ * saved with the logs that were sent before the call.<br><br>
+ * If <code>setLoggerConfig</code> wasn't called, the first log attempt will initialize a new logger with default configuration decided
+ * by the Rust core.
+ */
+public final class Logger {
+    // TODO: consider lazy loading the glide_rs library
+    static {
+        System.loadLibrary("glide_rs");
```

Comment:
can you move the static loading to the ffi.resolvers package? 
Example: https://github.com/Bit-Quill/glide-for-redis/blob/c998a2c366b524270a82c1148bfa477f7ab3318c/java/client/src/main/java/glide/ffi/resolvers/SocketListenerResolver.java

Reply:
I think you can remove this static loader - I don't think it doesn't anything here.